include/GcInfo/GcInfo.h

```diff
@@ -29,46 +29,103 @@
 class GcInfoAllocator;
 class GcInfoEncoder;
 
-/// \brief Per Function GcInfo
+// Allocation type information for a stack allocation.
+//
+// While some of the following information is deducible
+// from the Alloca instruction (ex: GcPointer/Aggregate)
+// we cache the information in the flag bits instead of
+// walking through the type every time.
+
+enum AllocaFlags {
+  Alloca_None = 0x0,
+  Alloca_GcPointer = 0x1,
+  Alloca_GcAggregate = 0x2,
+  Alloca_Pinned = 0x4,
+  Alloca_GsCookie = 0x8,
+  Alloca_SecurityObject = 0x10,
+  Alloca_GenericsContext = 0x20
+};
+
+// Per stack allocation GcInfo
+//
 // This Data-structure records the frame-offsets for certain special
 // allocations in each function generated by LLILC. This includes:
 //  -> Special Symbols: GsCookie, SecurityObject, GenericsContext
 //                      which must be identified to the runtime.
 //  -> Pinned pointers: Whose location must be reported to the runtime.
-//                      Pinned pointers are reported as untracked
-//  -> GC Aggregates:   Location of all GC-aggregates allocated on
-//                      the stack, also reported as untracked.
+//  -> GC Aggregates:   Location of all aggregates containing GC pointers
+//                      allocated on the stack
+//  -> GC Pointers:     GC Pointers allocated on the stack.
+//
+// Gc Pointers and GC aggregates are collectively referred to as GC Values.
 //
 // The stackmap generated via Statepoints only handles SSA values,
-// and therefore does not track pointers contained within GC-aggregates.
-// To circumvent this problem, we report all pointers within
-// GC-aggregates as untracked. The Gc-aggregate stack allocations are
-// known to be live throughout the function, because the reader marks
-// them as frame-escaped.
+// and therefore does not track pointers within stack allocated GC Values.
+//
+// To circumvent this problem, we report all pointers within stack allocated
+// GC Values as untracked.
+// These stack allocations are known to be live throughout the function,
+// because the reader marks them as frame-escaped.
+
+struct AllocaInfo {
+  int32_t Offset;
+  AllocaFlags Flags; // Combination of AllocaFlags, to be precise.
+  const char *getAllocTypeString() const;
+  bool isGcPointer() const { return ((Flags & Alloca_GcPointer) != 0); }
+  bool isGcAggregate() const { return ((Flags & Alloca_GcAggregate) != 0); }
+  bool isGcValue() const {
+    return ((Flags & (Alloca_GcPointer | Alloca_GcAggregate)) != 0);
+  }
```

Comment:
Since you have a flags enumeration you could add `GcValue = GcPointer | GcAggregate` there.